extensions/amp-next-page/1.0/service.js

```diff
@@ -401,45 +421,39 @@ export class NextPageService {
    * @return {!Page}
    */
   createHostPage() {
-    const doc = this.win_.document;
-    const {title, location} = doc;
+    const {title, location} = this.doc_;
     const {href: url} = location;
     const image =
-      parseSchemaImage(doc) || parseOgImage(doc) || parseFavicon(doc) || '';
+      parseSchemaImage(this.doc_) ||
```

Comment:
This smells like it should be a shared utility in `src` :)

Reply:
I have this https://github.com/ampproject/amphtml/pull/25978 for it

Reply:
How does #25978 relate? Seems like a larger refactor.

Reply:
It attempts to put all of these metadata (og tags, schema, favicon, etc) inside the doc info service

Reply:
What are the gains from moving into the service? Something other than API consolidation?

(not a blocker for any of this btw, just curious)

Reply:
API consolidation and caching of the attributes (read once) but let's move the discussion there